wvdecrypter/cdm/media/cdm/cdm_adapter.cc

```diff
@@ -504,10 +485,40 @@ cdm::Buffer* CdmAdapter::Allocate(uint32_t capacity)
   return client_->AllocateBuffer(capacity);
 }
 
+void CdmAdapter::timerfunc(CdmAdapter* adp, int64_t delay, void* context)
+{
+  {
+    std::unique_lock<std::mutex> lock(m_closeSessionMutex);
+    const bool isClosing = m_sessionClosingCond.wait_for(
+      lock, std::chrono::milliseconds(delay), [this] { return m_isClosingSession.load(); });
+    if (isClosing)
+      return;
+  }
+  adp->TimerExpired(context);
+}
+
 void CdmAdapter::SetTimer(int64_t delay_ms, void* context)
 {
   //LICENSERENEWAL
-  std::thread(timerfunc, shared_from_this(), delay_ms, context).detach();
+  // Clean up async tasks which have finished
+  {
+    std::lock_guard<std::mutex> lock(m_closeSessionMutex);
+    LOG::Log(SSD::SSDDEBUG, "VEC: %u", m_asyncTimerTasks.size());
```

Comment:
you have forgot to remove this and the other one below

Reply:
Sorry, fixed now. Need more sleep.